crates/lang_srv/src/analysis.rs

```diff
@@ -108,15 +135,68 @@ pub(crate) fn global_analysis(doc_info: DocInfo) -> Vec<AnalyzedDocument> {
         declarations_by_id: &mut declarations_by_id,
         typechecked: &mut typechecked,
         root_module: &mut root_module,
+        exposed_imports,
+        imports: &mut imports,
+        modules_info,
     };
 
     for (module_id, (path, source)) in sources {
-        documents.push(builder.build_document(path, source, module_id, doc_info.version));
+        let doc = builder.build_document(path, source, module_id, doc_info.version);
+        documents.push(doc);
     }
 
     documents
 }
 
+///Take the exposed imports from each module, lookup the symbol within that module's list of exposed symbols and then get the type info for that import
+///eg: `import {Task.{await}}`. `await` is an exposed_import, so we need to lookup its type info
+fn resolve_exposed_imports(
+    exposed_imports: MutMap<ModuleId, MutMap<Symbol, roc_region::all::Region>>,
+    exposes: &MutMap<ModuleId, Vec<(Symbol, Variable)>>,
+) -> HashMap<ModuleId, Vec<(Symbol, Variable)>> {
+    exposed_imports
+        .into_iter()
+        .map(|(module_id, symbols)| {
+            (
+                module_id,
+                symbols
+                    .into_iter()
+                    .filter_map(|(symbol, _)| {
+                        exposes.get(&module_id)?.iter().find(|(symb, _)| {
+                            //TODO this seems to not be comparing properly so we aren't getting any exposed imports
+                            symb == &symbol
+                        })
+                    })
+                    .cloned()
+                    .collect::<Vec<_>>(),
+            )
+        })
+        .collect()
+}
+///Transforms some of the raw data from the analysis into a state that is more useful during processes like completion
+fn make_modules_info(
+    exposes: MutMap<ModuleId, Vec<(Symbol, Variable)>>,
+    typechecked: &MutMap<ModuleId, CheckedModule>,
+) -> ModulesInfo {
+    //We wrap this in arc because later we will go through each module's imports and store the full list of symbols that each imported module exposes.
+    //eg: A imports B. B exposes [add, multiply, divide] and A will store a reference to that list.
```

Comment:
Why can't this be done now?

Reply:
I left it out because it's module specific and I thought it made sense to keep all of that in the AnalyzedDocumentBuilder. I also need the exported data later in the completion/formatting.rs file